pkg/server/handler/common.go

```diff
@@ -38,11 +39,19 @@ type AuthResponse struct {
 	LastSeenAt  *time.Time `json:"last_seen_at,omitempty"`
 }
 
-func NewAuthResponse(info skydb.AuthInfo, accessToken string) AuthResponse {
+func NewAuthResponse(info skydb.AuthInfo, authdata *skydb.AuthData, accessToken string) AuthResponse {
+	username := ""
+	email := ""
+
+	if authdata != nil {
+		username, _ = (*authdata)["username"].(string)
```

Comment:
The structure of `AuthData` would finally be dynamic, the `GetXXX()` methods would not exist.